snap/squashfs/squashfs.go

```diff
@@ -83,7 +86,13 @@ func (s *Snap) Install(targetPath, mountDir string) error {
 	// link(2) returns EPERM on filesystems that don't support
 	// hard links (like vfat), so checking the error here doesn't
 	// make sense vs just trying to copy it.
-	if err := os.Link(s.path, targetPath); err == nil {
+	if err := osLink(s.path, targetPath); err == nil {
+		return nil
+	}
+
+	// if the file is a seed, but the hardlink failed, symlinking it
+	// saves the copy (which in livecd is expensive) so try that next
+	if filepath.Dir(s.path) == dirs.SnapSeedDir && os.Symlink(s.path, targetPath) == nil {
```

Comment:
Will we ever end up retrying this and potentially try to create a symlink when one is already present?

Reply:
AFAIU it will return earlier after FilesAreEqual() check